projects/immer-data-strategy/src/lib/immer.data-strategy.ts

```diff
@@ -141,20 +141,25 @@ export class ImmerDataStrategy extends DataStrategy {
     private isConstructorArray(obj: any) {
         return obj.constructor === Array;
     }
+    
+    private isPropertyDefined(obj:any) {
+        return obj !== null && obj !== undefined;
+    }
 
     private extend(source: any, target: any, deep: boolean = true) {
         for (let prop in target) {
             if (target.hasOwnProperty(prop)) {
-                if (this.isConstructorArray(target[prop])) {
-                    source[prop] = [...target[prop]];
-                } else if (deep && this.isConstructorObject(target[prop])) {
-                    source[prop] = this.extend(source[prop], target[prop], deep);
+                if (this.isPropertyDefined(target[prop])) {
+                    if (this.isConstructorArray(target[prop])) {
+                        source[prop] = [...target[prop]];
+                    } else if (deep && this.isConstructorObject(target[prop])) {
+                        source[prop] = this.extend(source[prop], target[prop], deep);
+                    }
                 } else {
```

Comment:
Thats not good. it has 3 cases 1) either constructor 2) either object 3) either simple property. Since you have moved if to top that means if property is defined it can just be threated as constructor or object. 
Instead you need to leave as it was just add additional checks in constructor and obejcet if's

if(isPropertyDefined && isConstructor)
else if(isPropertyDefined && isConstructor)
else simple assigment